lib/import/hmcts_csv_parser.rb

```diff
@@ -0,0 +1,82 @@
+module Import
+  class HmctsCsvParser
+    CSV_HEADERS = %i(
+      publication_page_id
+      artefact_code
+      category
+      artefact_file_name
+      artefact_type
+      artefact_title
+      artefact_link
+      artefact_url
+      publication_type
+      publication_title
+      publication_summary
+      publication_body
+      published_before
+      has_images
+      alternative_format_email
+      policy
+      policy_areas
+      lead_organisations
+      supporting_organisations
+      excluded_nations
+      access_limiting
+      schedule_publishing
+      attachment_accessible
+      new_publication_url
+      new_attachment_url
+    ).freeze
+
+    PUBLICATION_TYPE_SLUGS = {
+      "Form" => "forms",
+      "Guidance" => "guidance",
+    }.freeze
+
+    def self.publications(path)
+      csv = CSV.read(path, headers: CSV_HEADERS)
+
+      Enumerator.new do |yielder|
+        page_id = nil
+        publication_details = {}
+
+        # Skip the header row
+        csv[1..-1].each do |row|
```

Comment:
If we use `CSV.read(path, headers: true, header_converters: :symbol)` then the first row is consumed and used as the headers definition and the strings are turned into lowercased symbols.  Meaning we can just do `csv.each` here. A tiny part of my brain wonders if the `[1..-1]` means we have to read in the whole file first to convert this to an array and then do the range thing, meaning we need a bunch of memory.  A different tiny part of my brain assumes that CSV and Enumerable do something smart so this isn't the case, but it's a smaller part of my brain and the first part is noisier.

Reply:
Hmm. I thought CSV did lazy-loading, but it looks like it stores everything in a big array. So the Enumerator's a bit unnecessary, and the `[1..-1]` presumably makes us loop over the data twice. 😞 

I don't think we can use `header_converters` here, unfortunately. Some of the columns have the same name, and others are quite verbose. I'm also expecting HMCTS to change a couple of them. It seemed less error-prone to just define them ourselves (even though we'll have to update the list if they add new columns).

Reply:
ah, shame about the headers thing :(